packages/common/src/providers/httpProvider.ts

```diff
@@ -0,0 +1,39 @@
+import Provider, { JsonRpcResponse } from './provider';
+import fetch from 'cross-fetch';
+
+export class HttpProvider implements Provider {
+    request: (
+        method: string,
+        params?: Record<string, unknown>
+    ) => Promise<JsonRpcResponse>;
+    nextId = 0;
```

Comment:
I believe using a `UUID` instead of an incrementing number as the `id` in the requests is better. If you reset the client using this class, then it will start using the same `id` values making it harder to search through requests based on a specific `id`.